src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -79,14 +79,16 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
     // Listing of the controllers that we have registered
     private registeredControllers = new Map<string, VSCodeNotebookController>();
     private selectedControllers = new Map<string, VSCodeNotebookController>();
-    private readonly allKernelConnections = new Set<KernelConnectionMetadata>();
```

Comment:
Remove `allKernelConnections` this was a duplicate of the values in `registeredControllers`.
and would go out of sync, hence changed to always derive the values from `registeredControllers`